# Deep Learning - Brain Tumor
*Group Members*:

|      | Student Name                  | Student ID |
|---   |-------------------------------|------------|
| 1    | Mahmoud Ahmed                 | 20221397   |
| 2    | Alex Adrián Santander Morales | 20220658   |
| 3    | Skander Chaabini              | 20221041   |
| 4    | Karim Miladi                  | 20220720   |
| 5    | Mohamed Ettaher Ben Slama     | 20221039   |

## Results Interpretation

Let's experiment with the best model (the one with the best validation accuracy):

**Concretely, our `EfficientNetB1` model with validation accuracy ~97% Yeaaah !!**

### Best model

Let's for last time evaluate best model on Testing dataset

In [117]:
loss_val, acc_val = effnet_model.evaluate(x=X_val, y=y_val)
loss_test, acc_test = effnet_model.evaluate(x=X_test, y=y_test)

41/41 [==============================] - 12s 297ms/step - loss: 0.0906 - accuracy: 0.9809


### Best model Accuracy

In [118]:
print (f"Validation Loss = {loss_val}")
print (f"Validation Accuracy = {acc_val}")

print (f"Test Loss = {loss_test}")
print (f"Test Accuracy = {acc_test}")

Validation Loss = 0.12626615166664124
Validation Accuracy = 0.9690781831741333
Test Loss = 0.09059450775384903
Test Accuracy = 0.9809305667877197


### Best model F1-Score

On validation dataset

In [105]:
y_val_prob = np.argmax(effnet_model.predict(X_val), axis=1)

f1_score_val = compute_f1_score(y_val, y_val_prob)
print(f"F1 score on Validation: {f1_score_val}")

54/54 [==============================] - 14s 257ms/step
F1 score: 0.9683734749590018


On testing dataset

In [106]:
y_test_prob = np.argmax(effnet_model.predict(X_test), axis=1)

f1_score_test = compute_f1_score(y_test, y_test_prob)
print(f"F1 score on Testing: {f1_score_test}")

41/41 [==============================] - 12s 287ms/step
F1 score on Validation: 0.9799833135561775


## Conclusion

Now, the model detects brain tumor with:

- `98.0%` accuracy on the test set.
- `0.97` f1 score on the test set.

These results are very good considering that the data is balanced.


Performance Table:

| Validation set | Test set |
|----------------|----------|
| Accuracy       | 96%      | 98% |
| F1 score       | 0.96     | 0.97 |

Hooray!